### PR TITLE
core: fix transaction hashes

### PR DESCRIPTION
**Pull request by AnnaShaleva**
We sign transactions using `(*Transaction).GetSignedPart`, so it should return serialised magic+hash.

### CODE REVIEW

File: pkg/core/block/block_test.go [hunk 1]
```diff
@@ -104,6 +104,7 @@ func TestHashBlockEqualsHashHeader(t *testing.T) {
 	assert.Equal(t, block.Hash(), block.Header.Hash())
 }
 
+/*
```
Comment (AnnaShaleva) [inline pkg/core/block/block_test.go:107]: Create an issue for these binary test-cases?

Comment (roman-khimov) [reply pkg/core/block/block_test.go:107]: Take something from C# privnet?

File: pkg/core/transaction/transaction_test.go [hunk 2]
```diff
@@ -130,6 +129,7 @@ func TestDecodingTxWithInvalidWitnessesNumber(t *testing.T) {
 	require.True(t, errors.Is(testserdes.DecodeBinary(data, new(Transaction)), ErrInvalidWitnessNum))
 }
 
+/*
 func TestUnmarshalNeoFSTX(t *testing.T) {
```
Comment (roman-khimov) [inline pkg/core/transaction/transaction_test.go:133]: It certainly can be fixed.

**Comment by roman-khimov** [comment]
Looks like I was distracted by something when doing #1829, this `updateHashes` part was supposed to be removed. Thanks for fixing it! Are we compatible now, have you tried mixed consensus?

**Comment by AnnaShaleva** [comment]
> Looks like I was distracted by something when doing #1829, this `updateHashes` part was supposed to be removed. Thanks for fixing it! Are we compatible now, have you tried mixed consensus?

The mixed consensus is not yet running because I found another bug in neo-go, I'll create a PR in a few minutes. Non-mixed consensus is running well for both C# and Go nodes (single- and four-nodes config is checked).

File: pkg/core/block/block_test.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/core/block/block_test.go]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**